### ISSUE TITLE
gpu-api vulkan: unable to render rgb24 png

### ISSUE THREAD

**Issue by cold-nagai** [mpv-player/mpv#12672]
### Important Information

Provide following Information:
<details><summary>mpv --version</summary>

```
mpv v0.36.0-577-g94b7584972 Copyright © 2000-2023 mpv/MPlayer/mplayer2 projects
 built on Oct 18 2023 16:06:57
libplacebo version: v6.338.0-31-gfd2addfb
FFmpeg version: n6.0
FFmpeg library versions:
   libavutil       58.2.100
   libavcodec      60.3.100
   libavformat     60.3.100
   libswscale      7.1.100
   libavfilter     9.3.100
   libswresample   4.10.100
```
</details>

- Arch Linux 6.5.7-arch1-1
- aur/mpv-git 0.36.0_577_g94b7584972-1 
- extra/hyprland 0.30.0-1
- aur/libplacebo-git 6.338.0.31.gfd2addfb-1

### Reproduction steps
 mpv --no-config --vo=gpu-next --gpu-api=vulkan

### Log file
```
[vo/gpu-next/libplacebo] Spent 17.151 ms loading cache
[vo/gpu-next] Assuming 60.012000 FPS for display sync.
[vo/gpu-next] reconfig to 400x400 rgb24 rgb/bt.709/srgb/full/display SP=1.000000 CL=mpeg1/jpeg crop=400x400+0+0
[vo/gpu-next/wayland] Reconfiguring!
[vo/gpu-next] Window size: 400x400 (Borders: l=0 t=0 r=0 b=0)
[vo/gpu-next] Video source: 400x400 (1:1)
[vo/gpu-next] Video display: (0, 0) 400x400 -> (0, 0) 400x400
[vo/gpu-next] Video scale: 1.000000/1.000000
[vo/gpu-next] OSD borders: l=0 t=0 r=0 b=0
[vo/gpu-next] Video borders: l=0 t=0 r=0 b=0
[vo/gpu-next/wayland] Handling resize on the vk side
[vo/gpu-next] Window size: 400x400 (Borders: l=0 t=0 r=0 b=0)
[vo/gpu-next] Video source: 400x400 (1:1)
[vo/gpu-next] Video display: (0, 0) 400x400 -> (0, 0) 400x400
[vo/gpu-next] Video scale: 1.000000/1.000000
[vo/gpu-next] OSD borders: l=0 t=0 r=0 b=0
[vo/gpu-next] Video borders: l=0 t=0 r=0 b=0
[vo/gpu-next/libplacebo] Rejecting texture format 'rgb8' due to misalignment: Row stride 1216 is not a clean multiple of texel size 3! This is likely an API usage bug.
[vo/gpu-next/libplacebo] Failed picking any compatible texture format for a plane!
[vo/gpu-next] Failed uploading frame!
[vo/gpu-next/libplacebo] Failed mapping frame id 0 with PTS 0.000000
[vo/gpu-next] Failed updating frames!
[vo/gpu-next/wayland] Given DND offer with mime type text/plain;charset=utf-8
[vo/gpu-next/wayland] Received a new DND offer. Releasing the previous offer.
[vo/gpu-next/wayland] Surface entered output BOE 0x06DF (0x32), scale = 1.000000, refresh rate = 60.012000 Hz
[vo/gpu-next/wayland] Obtained preferred scale, 1.000000, from the compositor.
[vo/gpu-next/wayland] Obtained preferred scale, 1.000000, from the compositor.
[vo/gpu-next/wayland] Resizing due to xdg from 400x400 to 1910x1035
[vo/gpu-next/wayland] Handling resize on the vk side
[vo/gpu-next] Window size: 1910x1035 (Borders: l=0 t=0 r=0 b=0)
[vo/gpu-next] Video source: 400x400 (1:1)
[vo/gpu-next] Video display: (0, 0) 400x400 -> (437, 0) 1035x1035
[vo/gpu-next] Video scale: 2.587500/2.587500
[vo/gpu-next] OSD borders: l=437 t=0 r=438 b=0
[vo/gpu-next] Video borders: l=437 t=0 r=438 b=0
[vo/gpu-next/wayland] Enabling idle inhibitor
[statusline] V: 00:00:00 / 00:00:00 (0%)
[vo/gpu-next] Failed updating frames!
[statusline] V: 00:00:00 / 00:00:00 (0%)
[vo/gpu-next] Failed updating frames!
[statusline] V: 00:00:00 / 00:00:00 (0%)
[statusline] V: 00:00:00 / 00:00:00 (0%)
[vo/gpu-next/wayland] Disabling the idle inhibitor
[vo/gpu-next/wayland] Deregistering output BOE 0x06DF (0x32)
```

**Comment by haasn** [comment]
Hmm, I can't seem to reproduce this error locally. That said, we have (in the meantime) gained a sliced upload path which can serve as a fallback for such oddly aligned buffers. So I think the proper solution here encompasses:

1. Figure out how and why you get misaligned buffers, and fix that (in mpv)
2. Enable sliced transfer path in all cases (inside libplacebo GPU backends)
3. Relax this alignment requirement

That said, I would prefer we do 1 before 2 and 3, since the latter will hide the issue but silently regress performance in the process.

**Comment by llyyr** [comment]
@cold-nagai is this on intel iGPU?

**Comment by cold-nagai** [comment]
> @cold-nagai is this on intel iGPU?

Yes



> 1. Figure out how and why you get misaligned buffers, and fix that (in mpv)

TL;DR I should experiment with mpv-build and checkouts after rgb48 fix and report back?
